event-hubs.api.md

```diff
@@ -0,0 +1,264 @@
+    enqueuedOn?: Date | number;
+    static fromEnqueuedTime(enqueuedOn: Date | number): EventPosition;
+    static fromOffset(offset: number): EventPosition;
+    static fromSequenceNumber(sequenceNumber: number, isInclusive?: boolean): EventPosition;
+    isInclusive: boolean;
+    static latest(): EventPosition;
+    offset?: number | "@latest";
+    sequenceNumber?: number;
+    }
+
+// @public
+export interface GetEventHubPropertiesOptions extends SpanOptions {
+    abortSignal?: AbortSignalLike;
+}
+
+// @public
+export interface GetPartitionIdsOptions extends SpanOptions {
+    abortSignal?: AbortSignalLike;
+}
+
+// @public
+export interface GetPartitionPropertiesOptions extends SpanOptions {
+    abortSignal?: AbortSignalLike;
+}
+
+// @public
+export interface InitializationContext extends PartitionContext {
+    setStartPosition(startPosition: EventPosition): void;
+}
+
+// @public
+export interface LastEnqueuedEventProperties {
+    enqueuedOn?: Date;
+    offset?: string;
+    retrievedOn?: Date;
+    sequenceNumber?: number;
+}
+
+export { MessagingError }
+
+// @public
+export interface PartitionContext {
+    consumerGroup: string;
+    eventHubName: string;
+    fullyQualifiedNamespace: string;
+    lastEnqueuedEventProperties?: LastEnqueuedEventProperties;
+    partitionId: string;
+    updateCheckpoint(eventData: ReceivedEventData): Promise<void>;
+}
+
+// @public
+export interface PartitionManager {
+    claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]>;
+    listCheckpoints(fullyQualifiedNamespace: string, eventHubName: string, consumerGroup: string): Promise<Checkpoint[]>;
+    listOwnership(fullyQualifiedNamespace: string, eventHubName: string, consumerGroup: string): Promise<PartitionOwnership[]>;
+    updateCheckpoint(checkpoint: Checkpoint): Promise<void>;
+}
+
+// @public
+export interface PartitionOwnership {
+    consumerGroup: string;
+    etag?: string;
+    eventHubName: string;
+    fullyQualifiedNamespace: string;
+    lastModifiedTimeInMs?: number;
+    ownerId: string;
+    partitionId: string;
+}
+
+// @public
+export interface PartitionProperties {
+    beginningSequenceNumber: number;
+    eventHubName: string;
+    lastEnqueuedOffset: number;
+    lastEnqueuedOnUtc: Date;
+    lastEnqueuedSequenceNumber: number;
+    partitionId: string;
+}
+
+// @public
+export type ProcessCloseHandler = (reason: CloseReason, context: PartitionContext) => Promise<void>;
+
+// @public
+export type ProcessErrorHandler = (error: Error, context: PartitionContext) => Promise<void>;
+
+// @public
+export type ProcessEventHandler = (receivedEvent: ReceivedEventData, context: PartitionContext) => Promise<void>;
+
+// @public
+export type ProcessInitializeHandler = (context: InitializationContext) => Promise<void>;
+
+// @public
+export interface ReceivedEventData {
+    body: any;
+    enqueuedTimeUtc: Date;
+    offset: number;
+    partitionKey: string | null;
+    properties?: {
```

Comment:
* rename properties to application_properties (check for consistency)
* last_enqueued_event_properties should be present?